client/state/notices/middleware.js

```diff
@@ -279,7 +279,7 @@ export const handlers = {
 	[ GUIDED_TRANSFER_HOST_DETAILS_SAVE_SUCCESS ]: dispatchSuccess( translate( 'Thanks for confirming those details!' ) ),
 	[ SITE_DELETE ]: onSiteDelete,
 	[ SITE_DELETE_FAILURE ]: onSiteDeleteFailure,
-	[ SITE_DELETE_SUCCESS ]: onSiteDeleteSuccess,
+	[ SITE_DELETE_RECEIVE ]: onSiteDeleteSuccess,
```

Comment:
Maybe we should change the handler name to match the action type, only because that seems to be the convention here?